### DISCUSSION TITLE
How have TypedController work with an union type?

### DISCUSSION THREAD

**Discussion by christopher-francisco** [react-hook-form/react-hook-form#2615] [Q&A]
I have a `CustomerFieldset` component that can be used in a form for both creating and updating it. The difference in their type is that the Edit one has an `id` prop, the creation doesn't.

However when I try to type it to work with both, it throws an error. I'm not sure what am I missing.

```typescript
// AddCustomer
const useFormMethods = useForm<AddCustomerFormValues>()
return <CustomerFieldset useFormMethods={useFormMethods} />

// EditCustomer
const useFormMethods = useForm<EditCustomerFormValues>()
return <CustomerFieldset useFormMethods={useFormMethods} />


// CustomerFieldset
type Props = { useFormMethods:  AddCustomerFormValues | EditCustomerFormValues }
...
const { control } = useFormMethods
const TypedController = useTypedController({ control });
```

I get the following Typescript error:

```
Type 'UseFormMethods<EditCustomerFormValues>' is not assignable to type 'UseFormMethods<AddCustomerFormValues | EditCustomerFormValues>'.
  Type 'AddCustomerFormValues | EditCustomerFormValues' is not assignable to type 'EditCustomerFormValues'.
    Property 'id' is missing in type 'CreateCustomerFormData' but required in type 'EditCustomerFormValues'.  TS2322

    86 |
  > 87 |             <CustomerFieldset useFormMethods={useFormMethods} />
``` 

What am I missing?

**Comment by bluebill1049** [comment]
Good question for @kotarella1110 🙏

**Reply by kotarella1110** [reply]
@christopher-francisco please provide codesandbox example!

**Reply by RenaudAubert** [reply]
Hi @kotarella1110 !
I know this discussion is a bit old but I'm still having this issue. Here's a [codesandbox](https://codesandbox.io/s/react-hook-form-useformmethods-typing-problems-i6m4b) where I have a typing error similar to what @christopher-francisco had.
This exemple uses `UseFormMethods` as is (which should accept any object) and with an union.

Let me know if you need more information or a clearer codesandbox.
